tools/workflow/clean_aws_resources.go

```diff
@@ -322,3 +328,67 @@ func destroyLoadBalancerResource() {
 		nextMarker = describeLoadBalancerOutputs.NextMarker
 	}
 }
+
+func destroyIamRoles() {
+	testSession, err := session.NewSession()
+
+	if err != nil {
+		log.Fatalf("Error creating session %v", err)
+	}
+
+	client := iam.New(testSession)
+
+	var rolesMarker *string
+
+	for {
+		lri := &iam.ListRolesInput{Marker: rolesMarker}
+		lro, err := client.ListRoles(lri)
+
+		if err != nil {
+			log.Fatalf("Failed to get roles because of %v", err)
+		}
+
+		for _, role := range lro.Roles {
+			var hasPrefix bool
+			for _, prefix := range roleNamePrefixes {
+				if hasPrefix = strings.HasPrefix(*role.RoleName, prefix); hasPrefix {
+					break
+				}
+			}
+			if hasPrefix && time.Now().UTC().Add(pastDayDeleteCalculation).After(*role.CreateDate) {
```

Comment:
We may want to also have it delete by date since last used

Reply:
For some reason this is always nil, but I can add it in on the off chance that it's available.

```go
if role.RoleLastUsed != nil {
  log.Printf("Last used %s %v", *role.RoleName, *role.RoleLastUsed.LastUsedDate)
}
```

Reply:
Prefer last used date instead of create date.

Reply:
It's nil for all of the roles in our account. I've added it to the conditional in case it's ever not nil.

Reply:
nil I think means not used

Reply:
It seems like a bug in the AWS SDK https://github.com/aws/aws-sdk/issues/24